src/fideslang/models.py

```diff
@@ -285,23 +298,150 @@ class MyDatasetField(DatasetFieldBase):
     )
 
 
-class DatasetField(DatasetFieldBase):
+EdgeDirection = Literal["from", "to"]
+
+
+class FidesDatasetReference(BaseModel):
+    """Reference to a field from another Collection"""
+
+    dataset: FidesKey
+    field: str
+    direction: Optional[EdgeDirection]
+
+
+class FidesMeta(BaseModel):
+    """Annotations used for query traversal"""
+
+    references: Optional[List[FidesDatasetReference]] = None
+    identity: Optional[str]
+    primary_key: Optional[bool]
+    data_type: Optional[str]
+    """Optionally specify the data type. Fides will attempt to cast values to this type when querying."""
+    length: Optional[PositiveInt]
+    """Optionally specify the allowable field length. Fides will not generate values that exceed this size."""
+    return_all_elements: Optional[bool]
+    """Optionally specify to query for the entire array if the array is an entrypoint into the node. Default is False."""
+    read_only: Optional[bool]
+    """Optionally specify if a field is read-only, meaning it can't be updated or deleted."""
+
+    @validator("data_type")
+    @classmethod
+    def valid_data_type(cls, value: Optional[str]) -> Optional[str]:
+        """Validate that all annotated data types exist in the taxonomy"""
+        return valid_data_type(value)
+
+
+class FidesopsMetaBackwardsCompat(BaseModel):
+    """Mixin to convert fidesops_meta to fides_meta for backwards compatibility
+    as we add DSR concepts to fideslang"""
+
+    def __init__(self, **data: Any) -> None:
```

Comment:
Instead of using `Any` is it possible to use these explicitly as the docstring has?
```suggestion
    def __init__(self, **data: Union[Dataset, DatasetCollection, DatasetField]) -> None:
```